bin/analyzeShearTest.py

```diff
@@ -0,0 +1,265 @@
+            catE1DevSum = 0.0
+            catE2DevSum = 0.0
+            catWeightSum = 0.0
+            sourceCat = butler.get("src", {'subfield':subfield, 'epoch':epoch})
+
+            #  These are constant values per subfield recorded by galsim
+            g1Key = sourceCat.getSchema().find("g1").getKey()
+            g2Key = sourceCat.getSchema().find("g2").getKey()
+
+            #   Set up the measurement keys, depending on type (moments or ellip)
+            if shape_type == "ellipticity":
+                e1Key = sourceCat.getSchema().find(shape_field + "e1").getKey()
+                e2Key = sourceCat.getSchema().find(shape_field + "e2").getKey()
+            else:
+                xxKey = sourceCat.getSchema().find(shape_field + "xx").getKey()
+                yyKey = sourceCat.getSchema().find(shape_field + "yy").getKey()
+                xyKey = sourceCat.getSchema().find(shape_field + "xy").getKey()
+
+            #   If there is a sigma field, get it.  If it also has a signal field
+            #   for signal-to-noise determination, get that too
+            if sigma_field == None:
+                sigmaKey = None
+                sigmaSignalKey = None
+            else:
+                sigmaKey = sourceCat.getSchema().find(sigma_field).getKey()
+                if sigma_signal_field == None:
+                    sigmaSignalKey = None
+                else:
+                    sigmaSignalKey = sourceCat.getSchema().find(sigma_signal_field).getKey()
+
+            #   Now loop through the catalog and summarize the ellipticity measurements
+            for source in sourceCat:
+                if shape_type == "ellipticity":
+                    e1 = source.get(e1Key)
+                    e2 = source.get(e2Key)
+                else:
+                    xx = source.get(xxKey)
+                    yy = source.get(yyKey)
+                    xy = source.get(xyKey)
+                    e1 = (xx - yy) / (xx + yy)
+                    e2 = 2 * xy / (xx + yy)
+                e = math.sqrt(e1 * e1 + e2 * e2)
+                g1 = source.get(g1Key)
+                g2 = source.get(g2Key)
+
+                #   Calculate the rotation in the range [-pi, pi], just to check the pairs
+                theta = math.atan2(e2,e1)/2.0
+                if theta >= math.pi:
+                    theta = theta - math.pi
+                if theta <= -math.pi:
+                    theta = theta + math.pi
+                if sigmaKey == None:
+                    weight = 1.0
+                else:
+                    #   Calculate the ellipticity error if fields are defined to do this
+                    #   Use it to calculate the variance, assuming a .25 shape noise
+                    sigma = source.get(sigmaKey)
+                    if not sigmaSignalKey == None:
+                        sigmaSignal = source.get(sigmaSignalKey)
+                        sigma = sigma * e / sigmaSignal
+                    weight = 1.0/(.25*.25 + sigma*sigma)
```

Comment:
Probably makes sense to make the shape noise contribution configurable as well.  On that note, I think I tend to see 0.3 more frequently than 0.25, but if you got this from James, it's fine to go with his recommendation.